src/Npgsql/TypeHandlers/CompositeHandlers/ByReference`1.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System.Diagnostics;
+
+namespace Npgsql.TypeHandlers.CompositeHandlers
+{
+    sealed class ByReference<T>
+    {
+        public T Value;
+
+        public ByReference(T value)
+        {
+            Debug.Assert(IsTrue);
+            Value = value;
+        }
+
+        public static readonly bool IsTrue = typeof(T).IsValueType;
```

Comment:
Had no better name in mind that that, and yes, it's confusing (:

Reply:
It would be good to check that there's an actual perf benefit to having this - maybe just putting `typeof(T).IsValueType` everywhere is already optimized by the JIT (if it's not it should be :)).

Otherwise I'd just call the property `IsValueType`...

Reply:
By the way, there is `RuntimeHelpers.IsReferenceOrContainsReferences<T>` which isn't available for all out targets.

Reply:
Re `RuntimeHelpers.IsReferenceOrContainsReferences<T>`, if this returns true on something that contains a reference field, then it wouldn't be good for us... A struct which contains a string is still a struct and should be treated as such.

Reply:
Checked, you're right. The previous version (`IsReference<T>`) was designed to say the same thing as `Type.IsValue`. Unfortunately, there is no documentation about it.